### DISCUSSION TITLE
How train a textcat with 1000 labels?

### DISCUSSION THREAD

**Discussion by info2000** [explosion/spaCy#8438] [Help: Best practices]
I want to categorize text with spacy, based on the textcat demo project, the json dataset should had all the labels on each item, like this:
```
    {"text": "Sorpresa: Ceferin comió con Piqué y Laporta",   "cats": {"AR": 0, "ES": 1, "PT": 0 } } 
    {"text": "Puado, de calabazas al Madrid a una renovación prioritaria","cats": {"AR": 0, "ES": 1, "PT": 0 } } 

```
But how should looks this json (or what is the right way to do) when have 1000 labels? 

Thanks

**Comment by polm** [comment]
There is no difference in data format between 1000 labels and 3 labels. You might have accuracy issues because the decision is probably harder, but you should be able to at least run training and see how it does.

Note that the JSON format is just a temporary format and not particularly important, the key thing is creating the docs that are saved to a DocBin. See the conversion scripts with the example projects to see how that works.

**Reply by info2000** [reply]
Thanks.
There's some textcat example with multiclasses?
```
text...., cat { content: "news", topic "energy" }
text...., cat { content: "blog", topic "political" }
```
or should be
`text...., cat { content_news: 1, topic_energy: 1 }`

I wrote in json model to better understand about what I'm asking

**Reply by polm** [reply]
To copy from your first post, the data should be more like this:

    "cats": {"AR": 0, "ES": 1, "PT": 0 }

The important thing is that each category is a key, and the value is 1 (has label) or 0 (does not have label). 

There's an example multilabel textcat project here:

https://github.com/explosion/projects/tree/v3/tutorials/textcat_goemotions

**Reply by info2000** [reply]
awesome sample
Thanks by your help
